lib/modules/datasource/metadata.ts

```diff
@@ -120,20 +121,15 @@ export function addMetaData(
   }
 
   if (
-    dep.changelogUrl?.includes('github.com') && // lgtm [js/incomplete-url-substring-sanitization]
+    dep.changelogUrl &&
+    detectPlatform(dep.changelogUrl) === 'github' &&
     !dep.sourceUrl
   ) {
     dep.sourceUrl = dep.changelogUrl;
   }
 
-  if (dep.homepage) {
-    const parsedHomePage = parseUrl(dep.homepage);
-    if (parsedHomePage?.hostname.includes('github')) {
-      if (!dep.sourceUrl) {
-        dep.sourceUrl = dep.homepage;
-      }
-      delete dep.homepage;
-    }
+  if (dep.homepage && !dep.sourceUrl) {
+    setSourceUrlToHomepage(dep);
   }
```

Comment:
I don't like this logic and naming. "if there's a home page but no source URL then set the source URl to the home page" is not logically correct.

Reply:
yes, bad naming 🤔

Reply:
the `shouldDeleteHomepage()` can be removed from inside `setSourceUrlToHomepage()`

Reply:
and keep a single place to check/delete the homepage